deployments.go

```diff
@@ -118,3 +118,81 @@ func (s *DeploymentsService) GetProjectDeployment(pid interface{}, deployment in
 
 	return d, resp, err
 }
+
+// CreateProjectDeploymentOptions represents the available
+// CreateProjectDeployment() options.
+//
+// GitLab API docs: https://docs.gitlab.com/ee/api/deployments.html#create-a-deployment
+type CreateProjectDeploymentOptions struct {
+	// The name of the environment to create the deployment for
+	Environment string `json:"environment"`
+
+	// The name of the branch or tag that is deployed
+	Ref string `json:"ref"`
+
+	// The SHA of the commit that is deployed
+	SHA string `json:"sha"`
+
+	// A boolean that indicates if the deployed ref is a tag (true) or not (false)
+	Tag bool `json:"tag"`
+
+	// The status of the deployment
+	Status BuildStateValue `json:"status"`
+}
```

Comment:
Please have a look at the other code in this package and try to make this fit in nicely. This means that all these options need to be pointers, they need a `url` tag and they all need the `omitempty` option. Additionally the field names and the link to the docs already give enough info, so we don't comment each field individually as well.

Reply:
> options need to be pointers

I mean, it's your repo and I don't want to be difficult, so if you really insist I can make this change, but my personal view is that since these fields are marked as required in the documentation, having them as pointers adds no value, just makes the API more difficult to use.

Reply:
I understand that, but at least it means that all option structs can be used in the same way. If we would differentiate between required and optional fields it would mean that you would need to look at each field to check if it expects a pointer to a type or just a type.

So in the end I think it does make it easier to work with as it's at least super consistent...

Reply:
Fair enough and no problem mate, you're the boss 🤷‍♂️

Reply:
Just a figure of speech :)